src/Symfony/Component/Form/Extension/DependencyInjection/DependencyInjectionExtension.php

```diff
@@ -58,8 +58,18 @@ public function getTypeExtensions($name)
             foreach ($this->typeExtensionServices[$name] as $serviceId => $extension) {
                 $extensions[] = $extension;
 
+                if (method_exists($extension, 'getExtendedTypes')) {
+                    $extendedTypes = array();
+
+                    foreach ($extension::getExtendedTypes() as $extendedType) {
+                        $extendedTypes[] = $extendedType;
+                    }
+                } else {
+                    $extendedTypes = array($extension->getExtendedType());
+                }
+
                 // validate result of getExtendedType() to ensure it is consistent with the service definition
-                if ($extension->getExtendedType() !== $name) {
+                if (!\in_array($name, $extendedTypes, true)) {
                     throw new InvalidArgumentException(sprintf('The extended type specified for the service "%s" does not match the actual extended type. Expected "%s", given "%s".', $serviceId, $name, $extension->getExtendedType()));
```

Comment:
`$extension->getExtendedType()` would need to be replaced by `$extendedTypes`. But as said above, the whole thing could be removed.